server/user/activity-tracker.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const { User } = require('../common/database');
+const debounce = require('lodash/debounce');
+const throttle = require('lodash/throttle');
+
+const toMiliseconds = min => min * 60000;
+
+class ActivityTracker {
+  constructor({ saveInterval, ttl }) {
+    this.tracked = {};
+    this.saveInterval = toMiliseconds(saveInterval);
+    this.ttl = toMiliseconds(ttl);
+  }
+
+  track(id) {
+    if (!this.tracked[id]) {
+      this.tracked[id] = {
+        save: throttle(() => this.save(id), this.saveInterval),
+        untrack: debounce(() => this.untrack(id), this.ttl)
+      };
+    }
+    this.tracked[id].lastActive = new Date();
+    this.tracked[id].untrack();
+    this.tracked[id].save();
+  }
+
+  untrack(id) {
+    if (!this.tracked[id]) return;
+    this.save(id).then(() => {
+      this.tracked[id].untrack.cancel();
+      this.tracked[id].save.cancel();
+      delete this.tracked[id];
```

Comment:
Back in the day `delete` was considered bad because it killed many perf optimizations. Instead what :neckbeard: -s used was:
```js
this.tracked[id] = null
```
Object keeps same shape and logic is still the same.